view/app-jarvis.js

```diff
@@ -58,6 +58,136 @@ app.controller('MainController', function($scope,$location,$rootScope,$http) {
 		}
 	});
 
+	$scope.getResponse = function(messageObj) {
+		const	data = 'username='+USER+'&message='+messageObj.message;
+		const index = $scope.messageStack.findIndex(obj => messageObj.fullDate === obj.fullDate && messageObj.text === obj.text);
+		$http({
+			url:URL+'/message',
+			method:'POST',
+			headers: {
+				'Content-Type': 'application/x-www-form-urlencoded'
+			},
+			data: data
+		}).then(resp => {
+			let res = (resp.data),
+				message = res['message'],
+				status = res['status'],
+				result = res['result'],
+				show = res['show'],
+				hrs2 = new Date().getHours(),
+				mins2 = new Date().getMinutes();
+			let resSuccess = false;
+			const resMessageObj = {
+				message: '',
+				sender: '',
+				time: '',
+				result: '',
+				show: false,
+				length: null
+			};
+			console.log(resMessageObj);
+			
+			if (status && message === 'here are the current weather conditions') {
+				resMessageObj.sender = 'jarvis-bot';
+				resMessageObj.time = String(new Date().getHours() + ':' + new Date().getMinutes());
+				resMessageObj.length = message.length;
+				resMessageObj.message = message;
+				resMessageObj.result = JSON.parse(result);
+				resSuccess = true;
```

Comment:
I see a lot of redundancy since this block of code is repeated in other conditions as well. Can we have a single function that handles this? Values could be simply passed to it and the response could be an object.